### ISSUE TITLE
Components: dismissible card close button is not accessible

### ISSUE THREAD

**Issue by bluefuton** [Automattic/wp-calypso#46378]
#### Steps to reproduce

The DismissibleCard component (which is used by Banner) uses a gridicon with an onClick event for its close button. 

It should use a `<button>` instead so it supports keyboard navigation and screen reader use.

#### What I expected

I can close a dismissible card with my keyboard and it is appropriately labelled for assistive technologies.

#### What happened instead

I can't reach the close button with my keyboard. It does not have a title.

#### Context / Source

Found while adding vote banner to Reader for NZ election: https://github.com/Automattic/wp-calypso/pull/46377#issuecomment-707392831.
